src/context/cart/cartContext.js

```diff
@@ -0,0 +1,99 @@
+import React, { createContext, useContext, useReducer } from "react";
+import { useEffect } from "react";
+import {
+  getCartApi,
+  addToCartApi,
+  updateQuantityApi,
+  removeFromCartApi,
+} from "../../services/cart_services";
+import { price } from "../../shared/utils/function";
+import { reducer } from "./cartReducer";
+
+const CartContext = createContext();
+
+export const CartContextProvider = ({ children }) => {
+  useEffect(() => {
+    if (status !== "IDLE") return;
+    (async function () {
+      dispatch({ type: "UPDATE_STATUS", payload: { status: "PENDING" } });
+      const { data, success, message } = await getCartApi();
+
+      if (success) {
+        dispatch({ type: "UPDATE_CART", payload: { cart: data.cart } });
+        dispatch({ type: "UPDATE_STATUS", payload: { status: "SUCCESS" } });
+      }
+      dispatch({ type: "UPDATE_STATUS", payload: { status: "FAILED" } });
+    })();
+  }, []);
+
+  const [{ cart, status }, dispatch] = useReducer(reducer, {
+    cart: [],
+    status: "IDLE",
+  });
+  const findItemInCart = (_id) => {
+    return cart.some((item) => {
+      return item._id === _id;
+    });
+  };
```

Comment:
fix: If possible move this function to utils

Reply:
This is not a utility function so I don't find any need to move it in utils .